src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.js';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const likedMessages = (id) => {
+    const newLikedData = chatData.map((chat) => {
+      if (chat.id === id) {
+        return { ...chat, liked: (chat.liked = !chat.liked) };
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(newLikedData);
+  };
+
+  const displayLikes = () => {
+    const likedData = chatData.filter((chat) => {
+      if (chat.liked) {
+        return chat;
+      }
```

Comment:
Be careful with the function we give to `filter`. For each item we iterate over (`chat`), we should return a truthy value if we want to include the item in the new list, and a falsy value if we want to exclude it. So we could write this as
```js
    chatData.filter(chat => chat.liked);
```

It's working as written, because any JS object is truthy (even an empty object), and if we reach the end of a function without an explicit return value, its result is `undefined` which is falsy. But we shouldn't return the item itself that we want in the filtered version. We should return a true/false value.